### ISSUE TITLE
not working

### ISSUE THREAD

**Issue by HansHammel** [TahirNaushad/Fiver.Mvc.FileUpload#1]

**Comment by kenneth-f-nielsen** [comment]
you need to add the wwwroot folder to the projects folder.

**Comment by HansHammel** [comment]
nope. done that. The part where the stream is written to disk is overwriting the local file. I think this wasn't testet or run on a local machine. The path concationation was wrong (last part is the local path that gets evaliuated first) and maybe other things...

**Comment by sonphnt** [comment]
Do you know why ZIP file get corrupted after downloading successfully? Thanks

**Comment by MVCLee** [comment]
I got this working with the following code for images and Word docx however, .pdf files don't work... they fail the first if statement (  if (file == null || file.Length == 0)   and the app stops on the next line
                 return Content("file not selected");

Does anyone have a fix to get .pdf files to upload? 

        public HomeController(IFileProvider fileProvider, IHostingEnvironment environment)
        {
            this.fileProvider = fileProvider;
            this.hostingEnvironment = environment;
        }
        public IActionResult Index()
        {
            return View();
        }
        [HttpPost]
        public async Task<IActionResult> UploadFile(IFormFile file)
        {
            if (file == null || file.Length == 0)          // .pdf files fail here...
                 return Content("file not selected");

            Guid guid = Guid.NewGuid();      // created guid
            var uploadedFileName = guid + Path.GetFileName(file.FileName);    // added guid to filename
            var path = Path.Combine(
                        //Directory.GetCurrentDirectory(), "wwwroot",       this line didn't work...
                        hostingEnvironment.WebRootPath + "\\documents\\", uploadedFileName);  // works

            using (var stream = new FileStream(path, FileMode.Create))
            {
                await file.CopyToAsync(stream);
            }
            return RedirectToAction("Files");
        }
